src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -6,7 +6,6 @@
     <Nullable>enable</Nullable>
     <TargetFrameworks>$(NetCoreAppCurrent);net5.0;netstandard1.1;netstandard1.3;net45;net46;netstandard2.0</TargetFrameworks>
     <ExcludeCurrentNetCoreAppFromPackage>true</ExcludeCurrentNetCoreAppFromPackage>
-    <AvoidRestoreCycleOnSelfReference>true</AvoidRestoreCycleOnSelfReference>
```

Comment:
We need to remove this hack anyhow for when we eventually convert this project to CSProj packing.

I looked into completely removing https://github.com/dotnet/runtime/blob/main/eng/AvoidRestoreCycleOnSelfReference.targets but it's still used here: https://github.com/dotnet/runtime/blob/28b9c72d25c3a58e85d08bc1ecf5e63a033dabe5/src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj#L6

And that can't be removed in the same way (we could refactor the project into a separate task project to fix it).